routemaster/state_machine/tests/test_visualisation.py

```diff
@@ -0,0 +1,91 @@
+from pathlib import Path
+
+import yaml
+
+from routemaster.config import load_config
+from routemaster.state_machine import nodes_for_cytoscape
+
+TEST_MACHINE_STATE_AS_NETWORK = [
+    {
+        'data': {'id': 'start'},
+        'classes': 'gate',
+    },
+    {
+        'data': {
+            'source': 'start',
+            'target': 'perform_action',
+        },
+    },
+    {
+        'data': {
+            'source': 'start',
+            'target': 'perform_alternate_action',
+        },
+    },
+    # We emit duplicate edges when the destination is duplicated; this seems to
+    # be fine though.
+    {
+        'data': {
+            'source': 'start',
+            'target': 'perform_action',
+        },
+    },
+    {
+        'data': {'id': 'perform_action'},
+        'classes': 'action',
+    },
+    {
+        'data': {
+            'source': 'perform_action',
+            'target': 'end',
+        },
+    },
+    {
+        'data': {'id': 'perform_alternate_action'},
+        'classes': 'action',
+    },
+    {
+        'data': {
+            'source': 'perform_alternate_action',
+            'target': 'end',
+        },
+    },
+    # We emit duplicate edges when the destination is duplicated; this seems to
+    # be fine though.
+    {
+        'data': {
+            'source': 'perform_alternate_action',
+            'target': 'start',
+        },
+    },
+    {
+        'data': {
+            'source': 'perform_alternate_action',
+            'target': 'end',
+        },
+    },
+    {
+        'data': {'id': 'end'},
+        'classes': 'gate',
+    },
+]
+
+
+def test_nodes_for_cytoscape(app):
+    nodes = nodes_for_cytoscape(app.config.state_machines['test_machine'])
+
+    assert nodes == TEST_MACHINE_STATE_AS_NETWORK
+
+
+def test_convert_example_to_network(app):
+    repo_root = Path(__file__).parent.parent.parent.parent
```

Comment:
This is not very pretty, and would be brittle to restructuring where this file is.

Reply:
a2f3e85